mqtt/src/main/scala/akka/stream/alpakka/mqtt/Mqtt.scala

```diff
@@ -79,26 +82,67 @@ final case class MqttConnectionSettings(
     socketFactory: Option[SSLSocketFactory] = None,
     cleanSession: Boolean = true,
-    will: Option[MqttMessage] = None
+    will: Option[MqttMessage] = None,
+    automaticReconnect: Boolean = false,
+    keepAliveInterval: FiniteDuration = 60.seconds,
+    connectionTimeout: FiniteDuration = 30.seconds,
+    maxInFlight: Int = 10,
+    mqttVersion: Int = MqttConnectOptions.MQTT_VERSION_3_1_1,
+    serverUris: Option[Seq[String]] = None,
+    sslHostnameVerifier: Option[HostnameVerifier] = None,
+    sslProperties: Option[Map[String, String]] = None
 ) {
-  def withBroker(broker: String) =
+  def withBroker(broker: String): MqttConnectionSettings =
     copy(broker = broker)
 
-  def withAuth(username: String, password: String) =
+  def withAuth(username: String, password: String): MqttConnectionSettings =
     copy(auth = Some((username, password)))
 
-  def withCleanSession(cleanSession: Boolean) =
+  def withCleanSession(cleanSession: Boolean): MqttConnectionSettings =
     copy(cleanSession = cleanSession)
 
-  def withWill(will: MqttMessage) =
+  def withWill(will: MqttMessage): MqttConnectionSettings =
     copy(will = Some(will))
 
   @deprecated("use a normal message instead of a will", "0.16")
-  def withWill(will: Will) =
+  def withWill(will: Will): MqttConnectionSettings =
     copy(will = Some(MqttMessage(will.message.topic, will.message.payload, Some(will.qos), will.retained)))
 
-  def withClientId(clientId: String) =
+  def withClientId(clientId: String): MqttConnectionSettings =
     copy(clientId = clientId)
+
+  def withAutomaticReconnect(automaticReconnect: Boolean): MqttConnectionSettings =
+    copy(automaticReconnect = automaticReconnect)
+
+  def withKeepAliveInterval(keepAliveInterval: FiniteDuration): MqttConnectionSettings =
+    copy(keepAliveInterval = keepAliveInterval)
+
+  def withKeepAliveInterval(keepAliveInterval: Int): MqttConnectionSettings =
+    withKeepAliveInterval(keepAliveInterval.seconds)
+
+  def withConnectionTimeout(connectionTimeout: FiniteDuration): MqttConnectionSettings =
+    copy(keepAliveInterval = keepAliveInterval)
+
+  def withConnectionTimeout(connectionTimeout: Int, unit: TimeUnit): MqttConnectionSettings =
+    copy(connectionTimeout = FiniteDuration(connectionTimeout, unit))
+
+  def withMaxInFlight(maxInFlight: Int): MqttConnectionSettings =
+    copy(maxInFlight = maxInFlight)
+
+  def withMqttVersion(mqttVersion: Int): MqttConnectionSettings =
+    copy(mqttVersion = mqttVersion)
+
+  def withServerUris(serverUris: Seq[String]): MqttConnectionSettings =
+    copy(serverUris = Some(serverUris))
+
+  @varargs def withServerUris(serverUris: String*): MqttConnectionSettings =
+    copy(serverUris = Some(serverUris.to[Seq]))
+
+  def withSslHostnameVerifier(sslHostnameVerifier: HostnameVerifier): MqttConnectionSettings =
+    copy(sslHostnameVerifier = Some(sslHostnameVerifier))
+
+  def withSslProperties(sslProperties: Map[String, String]): MqttConnectionSettings =
```

Comment:
Here it should support `java.util.Map`. The `withXxx` methods are considered Java-only API. Scala code will use `copy`. But yes, it is weird to go back and forth between Java and Scala collections.

Reply:
OK, now I understand the intent, thx. So I left only Java-related `withXxx` methods and addressed your comments.